content/troubleshooting/docs/reverse-proxy-sparkui.md

```diff
@@ -0,0 +1,290 @@
+```bash
+kubectl apply -f emr-eks-spark-example-01.yaml
+kubectl apply -f emr-eks-spark-example-02.yaml
+```
+
+3.Go to a web browser, then access their Spark Web UI while jobs are still running:
+
+```
+http://k8s-default-sparkui-2d325c0434-124141735.us-west-2.elb.amazonaws.com:80/sparkui/spark-example-01
+http://k8s-default-sparkui-2d325c0434-124141735.us-west-2.elb.amazonaws.com:80/sparkui/test-02
+```
+
+EKS Admin can provide the ALB endpoint address to users via this command: 
+```bash
+kubectl get ingress
+```
+ 
+## Launch EMR on EKS jobs via Job Run API
+
+1.Update the the environment variables in the sample job submission script:
+
+```bash
+export EMR_VIRTUAL_CLUSTER_NAME=YOUR_EMR_VIRTUAL_CLUSTER_NAME
+export AWS_REGION=YOUR_AWS_REGION
+export app_name=job-run-api
+
+export ACCOUNTID=$(aws sts get-caller-identity --query Account --output text)
+export VIRTUAL_CLUSTER_ID=$(aws emr-containers list-virtual-clusters --query "virtualClusters[?name == '$EMR_VIRTUAL_CLUSTER_NAME' && state == 'RUNNING'].id" --output text)
+export EMR_ROLE_ARN=arn:aws:iam::$ACCOUNTID:role/$EMR_VIRTUAL_CLUSTER_NAME-execution-role
+export S3BUCKET=$EMR_VIRTUAL_CLUSTER_NAME-$ACCOUNTID-$AWS_REGION
+
+aws emr-containers start-job-run \
+--virtual-cluster-id $VIRTUAL_CLUSTER_ID \
+--name $app_name \
+--execution-role-arn $EMR_ROLE_ARN \
+--release-label emr-7.1.0-latest \
+--job-driver '{
+"sparkSubmitJobDriver": {
+    "entryPoint": "local:///usr/lib/spark/examples/jars/spark-examples.jar", 
+    "entryPointArguments": ["100000"],
+    "sparkSubmitParameters": "--class org.apache.spark.examples.SparkPi --conf spark.executor.instances=1" }}' \
+--configuration-overrides '{
+"applicationConfiguration": [
+    {
+    "classification": "spark-defaults", 
+    "properties": {
+        "spark.ui.proxyBase": "/sparkui/`$app_name`",
+        "spark.ui.proxyRedirectUri": "/"
+    }}]}'
+```
+
+2.Once the job's driver pod is running, create a kubernetes service based on the driver pod name. Ensure its name contains the suffix **ui-svc**:
+```bash
+# query the driver pod name
+job_id=$(aws emr-containers list-job-runs --virtual-cluster-id $VIRTUAL_CLUSTER_ID --query "jobRuns[?name=='$app_name' && state=='RUNNING'].id" --output text)
+driver_pod_name=$(kubectl get po -n emr | grep $job_id-driver | awk '{print $1}')
+# create a k8s service
+kubectl expose po -n emr \
+--port=4040 \
+--target-port 4040 \
+--name=$app_name-ui-svc \
+$driver_pod_name
+```
+The SparkUI service looks like this:
+```bash
+kubectl get svc -n emr
+
+NAME                  TYPE      CLUSTER-IP  EXTERNAL-IP   PORT(S)  AGE
+job-run-api-ui-svc ClusterIP 10.100.233.186  <none>      4040/TCP   9s
+```
+
+3.Finally, access the SparkUI in this format:
+```
+http://<YOUR_INGRESS_ADDRESS>/sparkui/<app_name>
+```
+Admin can get the ingress address by the CLI:
+```bash
+kubectl get ingress
```

Comment:
We should mention this before the sparkUI above. Again, suggest to have expected output and what info to extract.